db/external_sst_file_ingestion_job.cc

```diff
@@ -317,32 +323,17 @@ Status ExternalSstFileIngestionJob::Prepare(
     }
   }
 
-  if (!status.ok()) {
-    // We failed, remove all files that we copied into the db
-    DeleteInternalFiles();
-  }
-
```

Comment:
Removing this `DeleteInternalFiles` call as it's redundant, more context in this comment thread: https://github.com/facebook/rocksdb/pull/12305#discussion_r1478720033